sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentModelOperationDetails.cs

```diff
@@ -4,38 +4,82 @@
 using System;
 using System.Collections.Generic;
 using System.Text.Json;
-using Azure.Core;
 
 namespace Azure.AI.FormRecognizer.DocumentAnalysis
 {
-    [CodeGenModel("GetOperationResponse")]
-    public partial class DocumentModelOperationDetails
+    /// <summary>
+    /// Details about a document model long-running operation.
+    /// </summary>
+    public class DocumentModelOperationDetails
     {
-        // This property is set by the DocumentAnalysisModelFactory when mocking this class.
-        private readonly ResponseError _mockError;
+        internal DocumentModelOperationDetails(GetOperationResponse response)
```

Comment:
Convenient constructor that converts from the generated `GetOperationResponse` to the manually written `DocumentModelOperationDetails`.